lib/logthrdest/logthrdestdrv.c

```diff
@@ -412,19 +439,34 @@ _perform_inserts(LogThreadedDestWorker *self)
 
   while (G_LIKELY(!self->owner->under_termination) && !self->suspended)
     {
+      ScratchBuffersMarker mark;
+      scratch_buffers_mark(&mark);
+
+      if (G_UNLIKELY(_flush_on_worker_partition_key_change_enabled(self)))
+        {
+          LogMessage *msg = log_queue_peek_head(self->queue);
+          if (!msg)
+            {
+              scratch_buffers_reclaim_marked(mark);
+              break;
+            }
+
+          if (_should_flush_due_to_partition_key_change(self, msg) && _perform_flush(self) != LTR_SUCCESS)
```

Comment:
I believe `LTR_EXPLICIT_ACK_MGMT` is another "successful" result code. Not sure if there is a use case of enabling this partition key induced flush and explicit ack management, though.